packages/angular/ssr/src/assets.ts

```diff
@@ -35,6 +35,16 @@ export class ServerAssets {
     return asset();
   }
 
+  /**
+   * Checks if a specific server-side asset exists.
+   *
+   * @param path - The path to the server asset.
+   * @returns A boolean indicating whether the asset exists.
+   */
+  hasServerAsset(path: string): boolean {
+    return this.manifest.assets.has(path);
+  }
```

Comment:
Consider: Could we update `getServerAsset` to return `Promise<string> | null` and drop `hasServerAsset`? I can't imagine many situations we'd want to check if an asset exists but not immediately read it.

Reply:
I considered this, but I am not too keen on this because we'd and up having to check for null and when `null` throw an error in a couple of other places.

Reply:
Ultimately up to you, I don't feel too strongly, just something to think about.